File: SDKs/googleapis.com/cloudfunctions/v1/python/src/sdk/models/shared/eventtrigger.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class EventTrigger:
    r"""EventTrigger
    Describes EventTrigger, used to request events be sent from another service.
    """
    
    event_type: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('eventType') }})
    failure_policy: Optional[FailurePolicy] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('failurePolicy') }})
    resource: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('resource') }})
    service: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('service') }})
    

File: SDKs/googleapis.com/cloudfunctions/v1/typescript/src/sdk/models/shared/eventtrigger.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FailurePolicy } from "./failurepolicy";



// EventTrigger
/** 
 * Describes EventTrigger, used to request events be sent from another service.
**/
export class EventTrigger extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=eventType" })
  eventType?: string;

  @SpeakeasyMetadata({ data: "json, name=failurePolicy" })
  failurePolicy?: FailurePolicy;

  @SpeakeasyMetadata({ data: "json, name=resource" })
  resource?: string;

  @SpeakeasyMetadata({ data: "json, name=service" })
  service?: string;
}
